src/te/schedule/schedule_lang.cc

```diff
@@ -267,10 +272,12 @@ Stage& Stage::fuse(const Array<IterVar>& axes, IterVar* p_target) {  // NOLINT(*
     IterVar singleton = IterVarNode::make(Range::make_by_min_extent(0, 1),
                                           Var("singleton", DataType::Int(32)), kDataPar);
     self->relations.push_back(SingletonNode::make(singleton));
-    ArrayNode* all_vars = self->all_iter_vars.CopyOnWrite();
-    ArrayNode* leaf_vars = self->leaf_iter_vars.CopyOnWrite();
-    all_vars->data.push_back(singleton);
-    leaf_vars->data.insert(leaf_vars->data.begin(), singleton);
+    Array<IterVar>& all_vars = self->all_iter_vars;
+    Array<IterVar>& leaf_vars = self->leaf_iter_vars;
+    all_vars.CopyOnWrite();
```

Comment:
CopyOnWrite are nolonger necessary?

Reply:
Good point! Originally I was thinking that `CopyOnWrite` might disable all raw pointers, so I added it for better safety. In this file I think it is okay to add it back :-)